### PR TITLE
fixes to struct servent and test for LookupService

### PR DESCRIPTION
**Pull request by ahgamut**
This PR adds necessary changes from the comments in #204, and adds tests for the `LookupService` pair of functions.
The tests run without any error.

### CODE REVIEW

**Comment by ahgamut** [comment]
@jart Unrelated to this PR: the `redbean_test.com` fails on my system (Debian 10 Linux).
This was happening even before #204. Is it related to any of my recent PRs? 

Attached screenshot:
2021-07-06_00-36-58_1366x768

**Comment by jart** [comment]
Definitely file an issue for that. -0x006c is MBEDTLS_ERR_ASN1_BUF_TOO_SMALL which would seem to indicate that it's trying to write a der elliptic curve certificate that's greater than 2**16 in length. I need to know how to reproduce that. For example, does it happen if you git clone and run `make`?

File: test/libc/dns/servicestxt_test.c [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+chargen		19/tcp		ttytst source\n\
+chargen		19/udp		ttytst source\n\
+ssh		    22/tcp		# SSH Remote Login Protocol";
+
+  ASSERT_NE(-1, (fd = creat("services", 0755)));
+  ASSERT_NE(-1, write(fd, sample, strlen(sample)));
+  ASSERT_NE(-1, close(fd));
+}
+
+TEST(LookupServicesByPort, GetNameWhenPortCorrect) {
```
Comment (jart) [inline test/libc/dns/servicestxt_test.c:50]: NICE

**Review by jart — APPROVED** [review]
LGTM

**Pull request merged**